src/config/spawn.rs

```diff
@@ -208,24 +243,19 @@ where
     }
 }
 
-impl<P: Process + Wait, S: Write> ShellSession<P, S> {
-    /// Sends the quit command to the shell.
-    /// Blocks until the shell process has exited.
+impl<P: Process + Wait, S: Write + Read + NonBlocking> ShellSession<P, S> {
+    /// Sends the quit command to the shell and attempts to terminate the shell process immediately.
     pub fn quit(&mut self) -> color_eyre::Result<()> {
-        if let Some(quit_command) = &self.quit_command {
-            let quit_command = quit_command.clone();
-            self.send_line(quit_command)
-                .wrap_err("error sending quit command to shell")?;
-        }
 
-        self.process
-            .wait(self.timeout)
-            .wrap_err("error waiting for shell to stop")?;
+        if let Some(quit_command) = self.quit_command.clone() {
+            self.send_line(&quit_command)?;
+        }
```

Comment:
So does waiting on the process to exit not work on macOS? Or do you think this is a zsh thing? Maybe the [`wait()` implementation](https://github.com/k9withabone/autocast/pull/10/files#diff-6570cbb3935efe426df4d55b84ad7bde974f19fa3c3dbe423bf01bc53a292e07R288) needs work?

Reply:
Thanks for the pointer to the `wait()` impl. I'll take another look at this to see if I can get something working, perhaps specific to macOS. I'm not particularly comfortable with the large-ish changes to `quit()` as they are. I'll take another shot at it.